contracts/mainnet/PolymorphRoot.sol

```diff
@@ -0,0 +1,205 @@
+
+    event PolymorphPriceChanged(uint256 newPolymorphPrice);
+    event MaxSupplyChanged(uint256 newMaxSupply);
+    event BulkBuyLimitChanged(uint256 newBulkBuyLimit);
+
+    constructor(
+        string memory name,
+        string memory symbol,
+        string memory baseURI,
+        address payable _daoAddress,
+        uint256 premintedTokensCount,
+        uint256 _baseGenomeChangePrice,
+        uint256 _polymorphPrice,
+        uint256 _maxSupply,
+        uint256 _randomizeGenomePrice,
+        uint256 _bulkBuyLimit,
+        string memory _arweaveAssetsJSON,
+        address _polymorphV1Address
+    )
+        PolymorphWithGeneChanger(
+            name,
+            symbol,
+            baseURI,
+            _daoAddress,
+            _baseGenomeChangePrice,
+            _randomizeGenomePrice,
+            _arweaveAssetsJSON
+        )
+    {
+        polymorphPrice = _polymorphPrice;
+        maxSupply = _maxSupply;
+        bulkBuyLimit = _bulkBuyLimit;
+        arweaveAssetsJSON = _arweaveAssetsJSON;
+        polymorphV1Contract = Polymorph(_polymorphV1Address);
+        geneGenerator.random();
+
+        _preMint(premintedTokensCount);
+    }
+
+    function _preMint(uint256 amountToMint) internal {
+        for (uint256 i = 0; i < amountToMint; i++) {
+            _tokenIdTracker.increment();
+            uint256 tokenId = _tokenIdTracker.current();
+            _genes[tokenId] = geneGenerator.random();
+            _mint(_msgSender(), tokenId);
+        }
+    }
+
+    function mint() public payable override nonReentrant {
+        require(_tokenIdTracker.current() < maxSupply, "Total supply reached");
+
+        _tokenIdTracker.increment();
+
+        uint256 tokenId = _tokenIdTracker.current();
+        _genes[tokenId] = geneGenerator.random();
+
+        (bool transferToDaoStatus, ) = daoAddress.call{value: polymorphPrice}(
+            ""
+        );
+        require(
+            transferToDaoStatus,
+            "Address: unable to send value, recipient may have reverted"
+        );
+
+        uint256 excessAmount = msg.value.sub(polymorphPrice);
+        if (excessAmount > 0) {
+            (bool returnExcessStatus, ) = _msgSender().call{
+                value: excessAmount
+            }("");
+            require(returnExcessStatus, "Failed to return excess.");
+        }
+
+        _mint(_msgSender(), tokenId);
+
+        emit TokenMinted(tokenId, _genes[tokenId]);
+        emit TokenMorphed(
+            tokenId,
+            0,
+            _genes[tokenId],
+            polymorphPrice,
+            PolymorphEventType.MINT
+        );
+    }
+
+    function burnAndMintNewPolymorph(uint256 tokenId) external nonReentrant {
+        require(_msgSender() == polymorphV1Contract.ownerOf(tokenId));
+
+        uint256 geneToTransfer = polymorphV1Contract.geneOf(tokenId);
+        polymorphV1Contract.burn(tokenId);
+
+        totalBurnedV1 = totalBurnedV1.add(1);
+        maxSupply = maxSupply.add(totalBurnedV1);
```

Comment:
This is my mistake, we should change it to:
`maxSupply = maxSupply.add(1);`
With the current implementation it will increment with all total burned every time :/

Reply:
Resolved